tests/models/idefics2/test_processing_idefics2.py

```diff
@@ -233,3 +261,17 @@ def test_apply_chat_template(self):
             "Assistant:"
         )
         self.assertEqual(rendered, expected_rendered)
+
+    def prepare_text_inputs(self, batched=False):
+        if batched:
+            return ["<image> lower newer", "<image> upper older longer string"]
+        return "<image> lower newer"
+
+    @require_vision
+    def prepare_image_inputs(self, batched=False):
+        """This function prepares a list of PIL images for testing"""
+        image_inputs = [np.random.randint(255, size=(3, 30, 400), dtype=np.uint8)]
+        image_inputs = [Image.fromarray(np.moveaxis(x, 0, -1)) for x in image_inputs]
```

Comment:
Recently `prepare_image_inputs` was moved out to the ProcessorTesterMixin to remove this repeated code. I'd instead remove this method and add batching logic to the mixin's method

Reply:
Note on that matter @yonigozlan there are some issues when using a rescale factor of -1 on a `uint8` 0-255 ranged image: it causes an overflow - easy to fix either in `rescale` or in this `prepare_image_inputs` (I'd go for the latter). Noticed that while trying to fix other broken tests!

Reply:
@amyeroberts I see, I'll add the batching logic to the mixin method by giving an int as you suggested above. However I think I will still have to override the prepare_image_inputs here, as Idefics2 expects batched images to be formatted as `[[images_for_text1], [images_for_text2]]`. 
@molbap That's weird I thought it would auto cast them to int16, do you have an example I can reproduce?

Reply:
OK, makes sense. Regarding the batching behaviour - is this not how all the processors expect the batching to occur given a variable number of images per sample? If not - then we probably have another round of work to standardize this across the processing classes

Reply:
It seems like not all models support batching in that sense... For example, Paligemma or Fuyu don't support nested images, so I'm not sure how/if they can deal with a variable number of images per sample. I will look into that more once I start working on the image-text-to-text pipeline again

Reply:
@yonigozlan OK, sounds good!

Reply:
The batching is indeed different for models with cross attention, as those require to pack and pad images with exactly `batch_size, num_images, ...` shape

For paligemma we recently did a fix to follow llava-like style, given the general architecture is +- same. Even though we can make idefics accept flat list of images (by inferring how many image tokens a each text has), I'm not sure it is intuitive. I remember we had similar discussion with @amyeroberts months ago, before accepting the `flat list` as possible input format. But now I am starting to think, it was a bad idea to push that into `main` hehe

But yeah, removing smth that users got used to is too much pain, so I guess we'll stick to accepting both formats...